packages/vscode-extensions/src/browser/vscode-extensions-service.ts

```diff
@@ -0,0 +1,152 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import * as showdown from 'showdown';
+import * as sanitize from 'sanitize-html';
+import { DisposableCollection, Emitter } from '@theia/core';
+import { VSCodeExtensionsAPI } from './vscode-extensions-api';
+import { SearchParam, VSCodeExtensionRaw, VSCodeExtension, VSCodeExtensionReviewList } from './vscode-extensions-types';
+import { VSCodeExtensionsModel } from './vscode-extensions-model';
+import { OpenerService, open } from '@theia/core/lib/browser';
+import { VSCodeExtensionUri, VSCodeExtensionDetailOpenerOptions } from './view/detail/vscode-extension-open-handler';
+import { PluginServer } from '@theia/plugin-ext';
+import { HostedPluginSupport } from '@theia/plugin-ext/lib/hosted/browser/hosted-plugin';
+
+export type ExtensionKeywords = string[];
+export const ExtensionKeywords = Symbol('ExtensionKeyword');
+
+// for now to test the extension one has to start the Registry Server via https://gitpod.io/#https://github.com/theia-ide/extension-registry
+// and copy then the workspace url plus '/api' to here
+export const API_URL = 'https://8080-c5484d3a-4b1a-4f66-81aa-c5fbc0d8ee0e.ws-eu01.gitpod-staging.com/api';
```

Comment:
We should host an instance somehow until we have the one under eclipse.org running.
Also this should become a preference.

Reply:
Not sure about preference, should a user be able to change it? I think it should be configurable for the prpduct